src/main/scala/org/broadinstitute/dsde/agora/server/ServerInitializer.scala

```diff
@@ -32,8 +36,18 @@ class ServerInitializer(val config: Config) extends LazyLogging {
   }
 
   private def startWebServiceActors() {
+    implicit val timeout = Timeout(5.seconds)
+    import scala.concurrent.ExecutionContext.Implicits.global
     val service = actorSystem.actorOf(ApiServiceActor.props, "agora-actor")
-    IO(Http) ! Http.Bind(service, interface = webserviceInterface, port = webservicePort)
+    (IO(Http) ? Http.Bind(service, interface = webserviceInterface, port = webservicePort)).onComplete {
+      case Success(Http.CommandFailed(failure)) =>
+        throw new Exception("could not bind to port: " + failure.toString)
+        actorSystem.shutdown()
+      case Failure(t) =>
+        throw new Exception("could not bind to port: " + t.toString)
+        actorSystem.shutdown()
+      case _ =>
+    }
```

Comment:
Per conversations with Jeff and Doug yesterday, I think I would still lean toward the synchronous `Await.result` way Zamboni does this (not necessarily something I'll say often :smile: )

I believe this code currently works because the actor system is currently the only daemon thread. As soon as the main thread and this daemon thread exit, the application can exit.  But as our applications mature and become more complex, we might have more daemon threads launched in which case this would only take down the actor system but could leave the app running but unresponsive to web requests.

Furthermore if the bind fails the actor system thread might get shut down quickly, but the main thread would continue with pointless but potentially expensive initialization (I'm thinking of a Hibernate SessionFactory as an example here).  So even with a single daemon thread, instead of seeing a fast failure if the bind fails we'd have to wait for all the initialization to run.